src/pages/home/sidebar/SidebarLinks.js

```diff
@@ -77,6 +77,13 @@ class SidebarLinks extends React.PureComponent {
         SidebarUtils.setIsSidebarLoadedReady();
         this.isSidebarLoaded = true;
 
+        // Eagerly set the locale on date-fns, it helps navigating to the report screen faster
+        InteractionManager.runAfterInteractions(() => {
+            requestAnimationFrame(() => {
+                this.props.updateLocale();
+            });
+        });
```

Comment:
This is a performance optimization. updateLocale doesn't really have to do anything with onyx but rather with the i18n library to load the correct localization file. Before it would run the first time the ReportScreen is mounted but it adds ~200ms delay to the animation of the screen. By running it asynchronously after sidebarlinks has been mounted (and is idle) we take advantage of an idle moment in device to load the file without interrupting the user.

Reply:
Yeah, everytime that function is called the locale is tried to be set, tbh I didn't want to refactor larger parts of the app, I was just tasked with improving the performance. The first call does load the locale, subsequent calls do nothing. I think that should be changed in a different PR, but for now, adding that call to sidebarlinks does improve the performance the first time the chat is opened.

Reply:
> can we tackle this in a separate issue ?

Yes we can